frontend/javascripts/oxalis/controller/combinations/segmentation_plane_controller.js

```diff
@@ -4,12 +4,106 @@ import Model from "oxalis/model";
+import createProgressCallback from "libs/progress_callback";
 import Store from "oxalis/store";
+import Toast from "libs/toast";
+import messages from "messages";
 
-function isosurfaceLeftClick(pos: Point2, plane: OrthoView, event: MouseEvent) {
+function segmentationLeftClick(pos: Point2, plane: OrthoView, event: MouseEvent) {
   if (!event.shiftKey) {
     return;
   }
+
+  if (event.ctrlKey) {
+    agglomerateSkeletonLeftClick(pos);
+  } else {
+    isosurfaceLeftClick(pos);
+  }
+}
+
+export async function agglomerateSkeletonLeftClick(pos: Point2) {
+  const segmentation = Model.getSegmentationLayer();
+  if (!segmentation) {
+    return;
+  }
+  const state = Store.getState();
+
+  const layerName =
+    segmentation.fallbackLayer != null ? segmentation.fallbackLayer : segmentation.name;
+
+  const { mappingName, mappingType } = state.temporaryConfiguration.activeMapping;
+  if (mappingName == null) {
+    Toast.error(messages["tracing.agglomerate_skeleton.no_mapping"]);
+    return;
+  }
+  if (mappingType !== "HDF5") {
+    Toast.error(messages["tracing.agglomerate_skeleton.no_agglomerate_file"]);
+    return;
+  }
+
+  const { renderMissingDataBlack } = state.datasetConfiguration;
+  const zoomStep = getRequestLogZoomStep(state);
+  const position = calculateGlobalPos(pos);
+
+  // While render missing data black is not active and there is no segmentation for the current zoom step,
+  // the segmentation of a higher zoom step is shown. Here we determine the the next zoom step of the
+  // displayed segmentation data to get the correct segment ids of the cell that was clicked on.
+  const usableZoomStep = renderMissingDataBlack
+    ? zoomStep
+    : segmentation.cube.getNextUsableZoomStepForPosition(position, zoomStep);
+
+  const cellId = segmentation.cube.getMappedDataValue(position, usableZoomStep);
+  if (cellId === 0) {
+    Toast.error(messages["tracing.agglomerate_skeleton.no_cell"]);
+    return;
+  }
+
+  const { dataset } = state;
+
+  const progressCallback = createProgressCallback({ pauseDelay: 100, successMessageDelay: 2000 });
+  const { hideFn } = await progressCallback(
+    false,
+    `Loading skeleton for agglomerate ${cellId} with mapping ${mappingName}`,
+  );
+
+  try {
+    const result = await getAgglomerateSkeleton(
+      dataset.dataStore.url,
+      dataset,
+      layerName,
+      mappingName,
+      cellId,
+    );
+
+    const nmlProtoBuffer = result;
+    const parsedTracing = parseProtoTracing(nmlProtoBuffer, "skeleton");
+
+    if (!parsedTracing.trees) {
+      // This check is only for flow to realize that we have a skeleton tracing
+      // on our hands.
+      throw new Error("Skeleton tracing doesn't contain trees");
+    }
+
+    Store.dispatch(
+      addTreesAndGroupsAction(
+        createMutableTreeMapFromTreeArray(parsedTracing.trees),
+        parsedTracing.treeGroups,
+      ),
+    );
+  } catch (e) {
+    console.error(e);
```

Comment:
Yes I think there will be, but the "expected" error cases (REST response fails or contains backend error) will show a message. I'm not sure whether we should add error handling for all other possible cases. When calling `parseProtoTracing`, for example, we also expect it to go right in the other spots where we call it. However, I've adapted the catch handler to only hide the notification and re-throw the error, so that the error is logged correctly (but still, no Toast is shown to the user, I think). If we want to show an error Toast to users every time an error is thrown, we should add a general error handler that does that. Currently, this wouldn't be trivial as for some errors a toast was already shown (request lib).
What do you think?

Reply:
Fully agree! I think the fact that the error wouldn't be logged to airbrake was the actual thing which bothered me subconsciously (but instead of identifying that I could only pinpoint the user-notification-thing). So, your recent change covers the issue nicely :+1: